compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -1191,12 +1191,12 @@ class Objects(using Context @constructorOnly):
 
     case ref: Ref =>
       val target = if needResolve then resolve(ref.klass, field) else field
-      if target.is(Flags.Lazy) then
-        given Scope = Env.emptyEnv(target.owner.asInstanceOf[ClassSymbol].primaryConstructor)
+      if target.is(Flags.Lazy) then // select a lazy field
         if ref.hasVal(target) then
           ref.valValue(target)
         else if target.hasSource then
           val rhs = target.defTree.asInstanceOf[ValDef].rhs
+          given Scope = Env.ofByName(target, rhs, ref, Env.NoEnv)
```

Comment:
Why is the `outerEnv` `NoEnv`? Can't the rhs of a lazy val refer to things from the enclosing scope of its definition site?

Reply:
OK, that makes sense. Thanks for the explanation. We can keep the current design.